src/compiler/visitor.js

```diff
@@ -3,46 +3,61 @@
  * the visitor pattern so we can map AST to functions
  */
 
+let noop = function() {};
+
 let visitor = {
   build(fns) {
-    let step = function(node) {
+    let step = function(node, index, context) {
       let type = node[0];
-      let extraArgs = Array.prototype.slice.call(arguments, 1);
```

Comment:
Maybe you can merge this, and then make the extraArgs change as you envision it.